akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKit.scala

```diff
@@ -215,6 +223,63 @@ final class ActorTestKit private[akka] (val name: String, val config: Config, se
    */
   def createTestProbe[M](name: String): TestProbe[M] = TestProbe(name)(system)
 
+  /**
+   * @return A test probe that is subscribed to unhandled messages from the system event bus. Subscription
+   *         will be completed and verified so any unhandled message after it will be caught by the probe.
+   */
+  def createUnhandledMessageProbe(): TestProbe[UnhandledMessage] = {
+    val probe = createTestProbe[Any]()
+    val narrowed = probe.ref.narrow[UnhandledMessage]
+    system.eventStream ! EventStream.Subscribe(narrowed)
+
+    // publish-receive roundtrip to verify we are subscribed before returning the probe
+    probe.awaitAssert {
+      val uh = UnhandledMessage(ActorTestKit.dummyMessage, probe.ref.toClassic, probe.ref.toClassic)
+      system.eventStream ! EventStream.Publish(uh)
+      if (probe.expectMessageType[UnhandledMessage].message != ActorTestKit.dummyMessage)
+        throw new RuntimeException("Not the expected unhandled")
+    }
+    probe.asInstanceOf[TestProbe[UnhandledMessage]]
+  }
+
+  /**
+   * @return A test probe that is subscribed to dead letters from the system event bus. Subscription
+   *         will be completed and verified so any dead letter after it will be caught by the probe.
+   */
+  def createDeadLetterMessageProbe(): TestProbe[DeadLetter] = {
+    val probe = createTestProbe[Any]()
+    val narrowed = probe.ref.narrow[DeadLetter]
+    system.eventStream ! EventStream.Subscribe(narrowed)
+
+    // publish-receive roundtrip to verify we are subscribed before returning the probe
+    probe.awaitAssert {
+      val dl = DeadLetter(ActorTestKit.dummyMessage, probe.ref.toClassic, probe.ref.toClassic)
+      system.eventStream ! EventStream.Publish(dl)
+      if (probe.expectMessageType[DeadLetter].message != ActorTestKit.dummyMessage)
+        throw new RuntimeException("Not the expected unhandled")
+    }
+    probe.asInstanceOf[TestProbe[DeadLetter]]
+  }
+
+  /**
+   * @return A test probe that is subscribed to dropped letters from the system event bus. Subscription
+   *         will be completed and verified so any dropped letter after it will be caught by the probe.
+   */
+  def createDroppedMessageProbe(): TestProbe[Dropped] = {
+    val probe = createTestProbe[Any]()
+    val narrowed = probe.ref.narrow[Dropped]
+    system.eventStream ! EventStream.Subscribe(narrowed)
+
+    // publish-receive roundtrip to verify we are subscribed before returning the probe
+    probe.awaitAssert {
+      val dl = Dropped(ActorTestKit.dummyMessage, "no reason", probe.ref.toClassic, probe.ref.toClassic)
+      system.eventStream ! EventStream.Publish(dl)
+      if (probe.expectMessageType[Dropped].message != ActorTestKit.dummyMessage)
+        throw new RuntimeException("Not the expected unhandled")
+    }
+    probe.asInstanceOf[TestProbe[Dropped]]
+  }
```

Comment:
Everything the same for these 3 apart for the published message. Should be possible to DRY.